PubSub/src/PubSubClient.php

```diff
@@ -459,6 +461,215 @@ function (array $snapshot) {
+     *     for details.
+     * @param array $options [optional] Configuration options
+     * @return Schema
+     */
+    public function createSchema($schemaId, $type, $definition, array $options = [])
+    {
+        $res = $this->connection->createSchema([
+            'parent' => SchemaServiceClient::projectName($this->projectId),
+            'schemaId' => $schemaId,
+            'type' => $type,
+            'definition' => $definition,
+        ] + $options);
+
+        return $this->schema($schemaId, $res);
+    }
+
+    /**
+     * Lists all schemas in the current project.
+     *
+     * Please note that the schemas returned will not contain the entire resource.
+     * If you need details on the full resource, call {@see Google\Cloud\PubSub\Schema::reload()}
+     * on the resource in question, or set `$options.view` to `FULL`.
+     *
+     * Example:
+     * ```
+     * $schemas = $pubsub->schemas();
+     * foreach ($schemas as $schema) {
+     *     $info = $schema->info();
+     *     echo $info['name']; // `projects/my-awesome-project/schemas/my-new-schema`
+     * }
+     * ```
+     *
+     * @see https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.schemas/list List Schemas
+     *
+     * @param array $options [optional] {
+     *     Configuration Options
+     *
+     *     @type string $view The set of Schema fields to return in the
+     *           response. If not set, returns Schemas with `name` and `type`,
+     *           but not `definition`. Set to `FULL` to retrieve all fields. For
+     *           allowed values, use constants defined on
+     *           {@see \Google\Cloud\PubSub\V1\SchemaView}.
+     *     @type int $pageSize Maximum number of results to return per
+     *           request.
+     *     @type int $resultLimit Limit the number of results returned in total.
+     *           **Defaults to** `0` (return all results).
+     *     @type string $pageToken A previously-returned page token used to
+     *           resume the loading of results from a specific point.
+     * }
+     * @return ItemIterator<Schema>
+     */
+    public function schemas(array $options = [])
+    {
+        $resultLimit = $this->pluck('resultLimit', $options, false);
+
+        return new ItemIterator(
+            new PageIterator(
+                function (array $schema) {
+                    $parts = SchemaServiceClient::parseName($schema['name'], 'schema');
+                    return $this->schema($parts['schema'], $schema);
+                },
+                [$this->connection, 'listSchemas'],
+                ['parent' => $this->formatName('project', $this->projectId)] + $options,
+                [
+                    'itemsKey' => 'schemas',
+                    'resultLimit' => $resultLimit
+                ]
+            )
+        );
+    }
+
+    /**
+     * Verify that a schema is valid before creating it.
```

Comment:
I would take out "before creating it." There is no requirement that one create the schema.